### PR TITLE
feat: add utility function to get supported chains from the Security Alerts API

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

This PR aims to include a new utility function to get supported chains from the Security Alerts API when enabled.

#### Related Repository
Refer to the [Security Alerts API repository](https://github.com/consensys-vertical-apps/va-mmcx-security-alerts-api) for more details.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25716?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2846

## **Manual testing steps**

1. Test blockaid regression

2. add the envs 
```shell
SECURITY_ALERTS_API_URL='https://security-alerts.dev-api.cx.metamask.io'
SECURITY_ALERTS_API_ENABLED='true'
```
- Go to test dapp and trigger on of the malicious signatures
- To verify in chrome go to dev tools > network. Search for `security-alerts` and find the call to the API service.

Existing PPOM logic should function as before, even with the above environment variables added, due to the fallback to the controller in the event of an error.

## **Screenshots/Recordings**

[supported-chains.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/60d0bcec-579d-42a7-8faf-e1be00dda1d8)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by jpuri — DISMISSED** [review]

File: app/scripts/lib/ppom/security-alerts-api.ts [hunk 1]
```diff
@@ -20,6 +22,10 @@ export async function validateWithSecurityAlertsAPI(
   return postRequest(endpoint, request);
 }
 
+export async function getSupportedChains(): Promise<Hex[]> {
```
Comment (matthewwalsh0) [inline app/scripts/lib/ppom/security-alerts-api.ts:25]: Same suggestions as the mobile PR 😄

File: app/scripts/lib/ppom/ppom-middleware.ts [hunk 2]
```diff
@@ -68,11 +66,12 @@ export function createPPOMMiddleware<
         preferencesController.store.getState()?.securityAlertsEnabled;
 
       const { chainId } = networkController.state.providerConfig;
+      const isSupportedChain = await isChainSupported(chainId);
 
       if (
         !securityAlertsEnabled ||
         !CONFIRMATION_METHODS.includes(req.method) ||
-        !SECURITY_PROVIDER_SUPPORTED_CHAIN_IDS.includes(chainId)
+        !isSupportedChain
```
Comment (matthewwalsh0) [inline app/scripts/lib/ppom/ppom-middleware.ts:74]: Do we also need to update this check within `app/scripts/lib/transaction/util.ts` or we could move it inside `validateRequestWithPPOM`?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: app/scripts/lib/ppom/ppom-middleware.ts (hunk 2 above)
Comment (vinistevam) [reply app/scripts/lib/ppom/ppom-middleware.ts:74]: I've updated the `validateSecurity, I think it is better to return early in this case. https://github.com/MetaMask/metamask-extension/pull/25716/commits/47cce7315573437388ce4c048f500957ada50944

File: app/scripts/lib/ppom/security-alerts-api.ts (hunk 1 above)
Comment (vinistevam) [reply app/scripts/lib/ppom/security-alerts-api.ts]: All suggestions applied the same as in mobile. https://github.com/MetaMask/metamask-extension/pull/25716/commits/36c4290737866c6255972bdc6b8e147b4ffacba0

File: app/scripts/lib/transaction/util.ts [hunk 3]
```diff
@@ -260,19 +262,19 @@ function validateSecurity(request: AddTransactionRequest) {
 
     const securityAlertId = generateSecurityAlertId();
 
-    validateRequestWithPPOM({
+    const securityAlertResponse = await validateRequestWithPPOM({
```
Comment (matthewwalsh0) [inline app/scripts/lib/transaction/util.ts:265]: This is all intentionally not awaited as we don't want to block the confirmation while the validation happens.

Hence why we immediately set the loading response, but handle any subsequent actual response with a `then`.

Maybe worth a comment to clarify that for future.

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: app/scripts/lib/transaction/util.ts (hunk 3 above)
Comment (vinistevam) [reply app/scripts/lib/transaction/util.ts]: Sorry, I reverted it and added a comment to avoid that happening again.

**Review by matthewwalsh0 — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**